### ISSUE TITLE
Is SQL Server planned in the future?

### ISSUE THREAD

**Issue by failshell** [cossacklabs/acra#470]
The list of supported databases is impressive. Is Microsoft SQL Server support planned? As well as the Azure SQL Server service?

**Comment by vixentael** [comment]
AcraServer works as proxy for [PostgreSQL wire protocol](https://www.postgresql.org/docs/14/protocol.html) and [MySQL wire protocol](https://dev.mysql.com/doc/internals/en/client-server-protocol.html) because these are open protocols and these database are SQL compliant. Overall, Acra supports all databases that works over these protocols. 

But not all SQL databases are SQL compliant. See [the SQL Compliance table](https://en.wikipedia.org/wiki/SQL_compliance).

AFAIK, Microsoft SQL Server uses its own proprietary SQL protocol and is not fully SQL compliant. 

AFAIK, Azure SQL used MySQL protocol under the hood, but they did a lot of changes to it. Azure SQL is not fully compliant with SQL protocol either (see https://docs.microsoft.com/en-us/azure/azure-sql/database/transact-sql-tsql-differences-sql-server) — the syntax and commands are different.

TLDR: these databases use different SQL flavours, mostly proprietary. Even if we teach AcraServer to support Microsoft SQL Server and Azure SQL Server, we couldn't guarantee the stability and maintainability of Acra because these SQL servers might change their behaviour in next version. That's definitely not something we plan to do in the nearest future for Acra Community Edition (custom flavors for Acra Enterprise Edition might be a case). 

To work with Microsoft SQL Server and Azure SQL Server we recommend using AcraTranslator (encryption-as-an-API) instead of AcraServer (SQL proxy). Application can store data in any database, just call AcraTranslator to encrypt/decrypt the data before/after storing. Please follow the [AcraTranslator description](https://github.com/cossacklabs/acra#protecting-data-in-any-file-storage-using-acratranslator) and [the guide](https://docs.cossacklabs.com/acra/acra-in-depth/architecture/acratranslator/).
